app/controllers/concerns/api/v3/sync_to_user.rb

```diff
@@ -1,20 +1,22 @@
 module Api::V3::SyncToUser
   extend ActiveSupport::Concern
+
   included do
-    def facility_group_records
-      current_facility_group
-        .send(model_name.name.underscore.pluralize.to_sym)
-        .with_discarded
+    def region_records
+      model_name.syncable_to_region(current_sync_region)
     end
 
     def current_facility_records
-      []
+      region_records
```

Comment:
Instead of an additional mixin called `PrioritisableByFacility` (or `PrioritisableByPatient` it should've been called now), we stick the default and most common behavior in `SyncToUser` for `current_facility_records`, `other_facility_records`.

Certain controllers like Facilities and Protocols can override it to avoid syncing scoped by `region_records` since we sync all of them all the time.

Notice that we've also changed the scope to be `patient` here instead of `facility`; this is more in-line with a proper patient-centric sync.

It used to be,

```ruby
current_facility_records.where(facility: current_facility).updated_since...
```

Which was: _give me all the blood_pressures for patients in the current_facility_group AND filter them by the current_facility_.

Now it's,

```ruby
region_records.where(patient: Patients.in_current_facility).updated_since...
```

Which is: _give me all the blood_pressures for patients in the current sync region AND filter them by patients in the current_facility_.

Reply:
This is potentially slower (we'll do some perf tests), but conceptually more correct.